project/frontend/src/App.js

```diff
@@ -1,236 +1,92 @@
 import "./App.css";
-import React, { useState } from "react";
-import { Route, Switch, Redirect, BrowserRouter } from "react-router-dom";
-
-import Explanation from "./containers/Explanation/Explanation";
-import SelectCompetition from "./containers/SelectCompetition/SelectCompetition";
+import React, { useState, useEffect } from "react";
+import { BrowserRouter } from "react-router-dom";
 
 import HeaderBar from "./components/HeaderBar/HeaderBar";
-import LandingPage from "./containers/LandingPage/LandingPage";
-import Dashboard from "./containers/Dashboard/Dashboard";
-import Competition from "./containers/Competition/Competition";
-import MySGonks from "./containers/MySGonks/MySGonks";
-
 import Layout from "./hoc/Layout/Layout";
+import PageRouter from "./hoc/PageRouter/PageRouter";
+
 import { AuthContext } from "./context/AuthContext";
 import { onAuthStateChange } from "./services/firebase";
-import Marketplace from "./containers/Marketplace/Marketplace";
 
-const NO_COMPETITION = 0;
+export const NO_COMPETITION = 0;
 
 function App() {
   const [user, setUser] = useState({ signedIn: false });
+  const [authStateReceived, setAuthStateReceived] = useState(false);
   const [compId, setCompId] = useState(0);
   const [competitionInfo, setCompetitionInfo] = useState({});
   const [loading, setLoading] = useState(false);
 
-  React.useEffect(() => {
-    const unsubscribe = onAuthStateChange(setUser);
+  useEffect(() => {
+    const unsubscribe = onAuthStateChange((userData) => {
+      setUser(userData);
+      setAuthStateReceived(true);
+    });
     return () => {
       unsubscribe();
     };
   }, []);
 
-  React.useEffect(() => {
+  useEffect(() => {
     const parsedCompId = Number(localStorage.getItem("compId") || 0);
     setCompId(parsedCompId);
   }, []);
 
-  if (user.signedIn && !user.id) {
-    //TODO: fetch for id... with useremail param... once backend is ready
-    //for now id hardcoded to 123
-    setUser((prevState) => {
+  const fetchCompetitionInfo = async (fetchCall, stateKey) => {
+    try {
+      const data = await fetch(fetchCall).then((response) => response.json());
       return {
-        ...prevState,
-        id: 1,
+        [stateKey]: data,
       };
-    });
-  }
+    } catch (e) {
+      return {
+        [stateKey]: undefined,
+      };
+    }
+  };
 
-  const fetchAndUpdateCompetitionInfo = async (fetchCall, stateKey) => {
-    return fetch(fetchCall)
-      .then((response) => response.json())
-      .then((data) => {
-        setCompetitionInfo((prevState) => {
-          return {
-            ...prevState,
-            [stateKey]: data,
-          };
-        });
-      });
+  const isReadyForDataFetch = () => {
+    if (!user.signedIn) return false;
+    if (user.id === undefined) return false;
```

Comment:
Maybe we should check if `user.id` is 0, since we don't expect that either

Reply:
Done.